File: backend/secureattend-backend/src/main/java/com/secureattend/domain/CourseClass.java
package com.secureattend.domain;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Document(collection = "classes")
public class CourseClass {
    @Id
    private String id;
    private String code;
    private String title;
    private String description;
    private String professorId;
    private List<String> taIds = new ArrayList<>();
    private List<String> studentIds = new ArrayList<>();
    private String semester;
    private Integer credits;
    private String schedule;
    private String location;
    private Double latitude;
    private Double longitude;
    private String wifiSSID;
    private Instant createdAt = Instant.now();
    private boolean active = true;

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    
    public String getCode() { return code; }
    public void setCode(String code) { this.code = code; }
    
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    
    public String getProfessorId() { return professorId; }
    public void setProfessorId(String professorId) { this.professorId = professorId; }
    
    public List<String> getTaIds() { return taIds; }
    public void setTaIds(List<String> taIds) { this.taIds = taIds; }
    
    public List<String> getStudentIds() { return studentIds; }
    public void setStudentIds(List<String> studentIds) { this.studentIds = studentIds; }
    
    public String getSemester() { return semester; }
    public void setSemester(String semester) { this.semester = semester; }
    
    public Integer getCredits() { return credits; }
    public void setCredits(Integer credits) { this.credits = credits; }
    
    public String getSchedule() { return schedule; }
    public void setSchedule(String schedule) { this.schedule = schedule; }
    
    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }
    
    public Double getLatitude() { return latitude; }
    public void setLatitude(Double latitude) { this.latitude = latitude; }
    
    public Double getLongitude() { return longitude; }
    public void setLongitude(Double longitude) { this.longitude = longitude; }
    
    public String getWifiSSID() { return wifiSSID; }
    public void setWifiSSID(String wifiSSID) { this.wifiSSID = wifiSSID; }
    
    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }
    
    public boolean isActive() { return active; }
    public void setActive(boolean active) { this.active = active; }
}

File: backend/src/models/CourseClass.js
import mongoose from 'mongoose';
import { randomUUID } from 'crypto';

const courseClassSchema = new mongoose.Schema(
  {
    _id: { type: String, default: () => randomUUID() },
    code: String,
    title: String,
    description: String,
    professorId: String,
    taIds: { type: [String], default: [] },
    studentIds: { type: [String], default: [] },
    semester: String,
    credits: Number,
    schedule: String,
    location: String,
    latitude: Number,
    longitude: Number,
    wifiSSID: String,
    createdAt: { type: Date, default: Date.now },
    active: { type: Boolean, default: true },
  },
  {
    collection: 'classes',
    versionKey: false,
    toJSON: {
      transform(_doc, ret) {
        ret.id = ret._id;
        delete ret._id;
        return ret;
      },
    },
  }
);

courseClassSchema.index({ professorId: 1 });
courseClassSchema.index({ studentIds: 1 });
courseClassSchema.index({ taIds: 1 });

export const CourseClass = mongoose.model('CourseClass', courseClassSchema);
